src/rpc/net.cpp

```diff
@@ -248,7 +248,16 @@ UniValue disconnectnode(const JSONRPCRequest& request)
     if(!g_connman)
         throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Peer-to-peer functionality missing or disabled");
 
-    bool ret = g_connman->DisconnectNode(request.params[0].get_str());
+    std::string node = (request.params[0].get_str());
+    bool ret;
+    if (std::all_of(node.begin(), node.end(), ::isdigit)) {
```

Comment:
Concept ACK, but please make this API explicit.

- APIs that switch based on magic heuristics on how a value 'looks'. Initially this seems user friendly, but it will quickly grow to maintainable, insecure monsters of obscure rules. It should be made completely explicit.

- Also: do not encode integers as strings and use functions like std::stoi. Just use .get_int() if you need an int value and leave the encoding of values up to the RPC mechanism.

The most straightforward way would be to just add a function `disconnectnodebyid`.